flink-ml-lib/src/test/java/org/apache/flink/ml/feature/MinHashLSHTest.java

```diff
@@ -0,0 +1,452 @@
+        List<Row> results = IteratorUtils.toList(tEnv.toDataStream(output).executeAndCollect());
+        compareResultCollections(
+                expected,
+                results,
+                (d0, d1) -> {
+                    DenseVectorArrayComparator denseVectorArrayComparator =
+                            new DenseVectorArrayComparator();
+                    return denseVectorArrayComparator.compare(d0.getFieldAs(0), d1.getFieldAs(0));
+                });
+    }
+
+    @Test
+    public void testHashFunction() {
+        MinHashLSHModelData lshModelData =
+                new MinHashLSHModelData(3, 1, new int[] {0, 1, 3}, new int[] {1, 2, 0});
+        Vector vec = Vectors.sparse(10, new int[] {2, 3, 5, 7}, new double[] {1., 1., 1., 1.});
+        DenseVector[] result = lshModelData.hashFunction(vec);
+        Assert.assertEquals(3, result.length);
+        Assert.assertEquals(Vectors.dense(1.), result[0]);
+        Assert.assertEquals(Vectors.dense(5.), result[1]);
+        Assert.assertEquals(Vectors.dense(9.), result[2]);
+    }
+
+    @Test
+    public void testHashFunctionEqualWithSparseDenseVector() {
+        // Use randomly generate coefficients, so that the hash values are not always from the least
+        // non-zero index.
+        MinHashLSHModelData lshModelData = MinHashLSHModelData.generateModelData(3, 1, 10, 2022L);
+        new MinHashLSHModelData(3, 1, new int[] {0, 1, 3}, new int[] {1, 2, 0});
+        Vector vec = Vectors.sparse(10, new int[] {2, 3, 5, 7}, new double[] {1., 1., 1., 1.});
+        DenseVector[] denseResults = lshModelData.hashFunction(vec.toDense());
+        DenseVector[] sparseResults = lshModelData.hashFunction(vec.toSparse());
+        Assert.assertArrayEquals(denseResults, sparseResults);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testHashFunctionWithEmptyVector() {
+        MinHashLSHModelData lshModelData =
+                new MinHashLSHModelData(3, 1, new int[] {0, 1, 3}, new int[] {1, 2, 0});
+        Vector vec = Vectors.sparse(10, new int[] {}, new double[] {});
+        lshModelData.hashFunction(vec);
+    }
+
+    @Test
+    public void testParam() {
+        MinHashLSH lsh = new MinHashLSH();
+        Assert.assertEquals("input", lsh.getInputCol());
+        Assert.assertEquals("output", lsh.getOutputCol());
+        Assert.assertEquals(MinHashLSH.class.getName().hashCode(), lsh.getSeed());
+        Assert.assertEquals(1, (int) lsh.getNumHashTables());
+        Assert.assertEquals(1, (int) lsh.getNumHashFunctionsPerTable());
+        lsh.setInputCol("vec")
+                .setOutputCol("hashes")
+                .setSeed(2022L)
+                .setNumHashTables(3)
+                .setNumHashFunctionsPerTable(4);
+        Assert.assertEquals("vec", lsh.getInputCol());
+        Assert.assertEquals("hashes", lsh.getOutputCol());
+        Assert.assertEquals(2022L, lsh.getSeed());
+        Assert.assertEquals(3, (int) lsh.getNumHashTables());
```

Comment:
We can avoid this casting here by letting `getNumHashTables()` return a `int`.

Same for `getNumHashFunctionsPerTable()`